### PR TITLE
fix(ml,#8052): reframe 2.8-PAC interp table to match deterministic output

### PR DESCRIPTION
**Pull request by jsboige**
Grain: MED/notebook-python-audit-claims — lane myia-po-2024:CoursIA-2 — prev: DEEP/notebook-dotnet (c.847 yielded)

## What

Fix `2.8-Theorie-PAC.ipynb` cell md[15] (the §4 "Interprétation : la richesse se paie en exemples") whose interpretation table **mischaracterizes its own deterministic output** (code[14]). The committed numbers are correct and reproducible — the *prose* around them was wrong. See #8052.

## The mismatch (re-executed firsthand, C846-L2)

code[14] output (committed, `make_moons(n=400, noise=0.25, random_state=42)` + `train_test_split(train_size=40, random_state=42)` — deterministic). I re-executed firsthand: **identical** to committed.

```
degre 1 : train = 0.775 | test = 0.831 | ecart = -0.056
degre 3 : train = 0.825 | test = 0.872 | ecart = -0.047
degre 5 : train = 0.925 | test = 0.914 | ecart = 0.011
degre 7 : train = 0.925 | test = 0.897 | ecart = 0.028
```

md[15] claimed:

| Claim | Reality (from output) |
|-------|----------------------|
| Degré 1 : "train et test **proches**, sous-apprentissage" | Degree 1 has the **largest** \|écart\| (0.056), and it is **negative** — test (0.831) > train (0.775). Not "proches", not a clean underfit. |
| "l'écart `train - test` **s'élargit**" | The *signed* gap does grow, but it **flips sign** at degree 5: −0.056 → −0.047 → +0.011 → +0.028. The *absolute* gap is **largest at degree 1**, not monotonic widening. |
| "5-7 ... écart train-test **marqué**" | Degree 5's gap is **0.011 (the smallest)**. Only degree 7 shows a (modest) positive gap. |
| train "grimpe" | Climbs then **plateaus at 0.925** for both degree 5 and 7. |

## Fix (markdown-only, byte-surgical)

Reframed md[15] to read the data honestly:
- **Constat** rewritten: train climbs monotonically to a 0.925 plateau; the signed `train - test` gap does not simply "s'élargir" — it **changes sign** (negative at low degree = test beats train, a small-`m=40` artifact; positive only at high degree).
- The 2-row hand-wavy table (`1` / `5-7`) replaced by a **4-row table grounded in the 4 actual degrees**, each row citing the real train/test/écart + a one-line honest lecture.
- Key correction: degree 1 is **not** a "proche/sous-apprentissage" regime — it has the largest gap, inverted (test > train). Degree 5 = generalization optimum (test peaks 0.914). Degree 7 = where genuine overfit finally appears (test retreats 0.914 → 0.897).
- Leçon tightened: "une classe plus riche *finit par* généraliser moins bien" (test recule degré 5→7) — the original absolute "généralise moins bien" was true only past degree 5.

### Scope (byte-surgical — C839-L1)

Only **md[15]** changed (+703 bytes). All 29 other cells **byte-identical to origin/main** (round-trip guard: `json.dumps(indent=1)` verified byte-identical pre-edit; exactly 1 cell changed post-edit, confirmed by per-cell diff).

**code[14] untouched** — it still carries its real committed output (ec=7, outputs=2). Its plot title string `"... ecart grandissant"` remains: consistent with the reframe (the *signed* gap does grow −0.056→+0.028), and changing a code string would force a re-exec (C.2) outside this grain's scope. **Markdown-only → C.2 exception** (no re-exec needed; outputs unchanged).

## Validation (firsthand)

- `nbformat.validate`: OK
- H.3 pre-commit: 0 violations (10 code cells, all `execution_count`/`outputs` present)
- C.1: clean (no `raise NotImplementedError` / `assert False` / `1/0`)
- code[14] output re-executed firsthand (seed=42) → **byte-identical to committed** (deterministic).

## Out of scope

- code[14] plot title `"ecart grandissant"`: defensible (signed gap grows), left as-is. A future grain could sharpen it to "écart signé croissant" if desired.

See #8052

### CODE REVIEW

**Review by clusterManager-Myia — COMMENTED** [review]
**[NanoClaw] structural review** — `fix(ml,#8052): reframe 2.8-PAC interp table to match deterministic output` (head `dd551a74`, +8/-6, 1f `MyIA.AI.Notebooks/ML/DataScienceWithAgents/02-ML-Cours/2.8-Theorie-PAC.ipynb`, cell idx 15 markdown, author jsboige ≠ myia-ai-01 → 0 self-review conflict). Structural review (cell-diff isolate, no full patch — glm-5.2 250k anti-thrash).

**LGTM deep** — audit-fix **PROUVÉ firsthand** : la nouvelle table interp est **byte-exact contre la sortie du code cell idx 14 (ec=7)** committée, et le reframing est **conceptuellement supérieur** (corrige une demi-vérité qui masquait le phénomène le plus intéressant).

**Code cell idx 14 (ec=7) émet littéralement** :
```
degre 1 : train = 0.775 | test = 0.831 | ecart = -0.056
degre 3 : train = 0.825 | test = 0.872 | ecart = -0.047
degre 5 : train = 0.925 | test = 0.914 | ecart = 0.011
degre 7 : train = 0.925 | test = 0.897 | ecart = 0.028
```

**Nouvelle table markdown = byte-exact ces 4 lignes** (train/test/écart) : deg 1 (0,775/0,831/-0,056), deg 3 (0,825/0,872/-0,047), deg 5 (0,925/0,914/+0,011), deg 7 (0,925/0,897/+0,028) ✓. Aucune fabrication.

**Reframing conceptuellement SOUND** : l'ancienne prose disait « l'écart train-test s'élargit avec le degré = surapprentissage » — **demi-vérité** qui lisait les écarts en valeur absolue et **masquait l'inversion de signe**. Or le phénomène empirique réel le plus riche ici est précisément : (a) **écart NÉGATIF aux bas degrés** (test *dépasse* train, 0,831 vs 0,775 au deg 1) = artefact du petit `m=40` (l'échantillon test, par chance, plus facile), PAS un sous-apprentissage « propre » ; (b) **l'écart change de signe au deg 5** (+0,011), reste minime au point d'être l'**optimum de généralisation** (test maximal 0,914) ; (c) **le surapprentissage n'apparaît qu'au deg 7** (+0,028, test recule 0,914→0,897). La nouvelle prose documente honnêtement ces 3 phases (inversion artefact → optimum → surapprentissage tardif) au lieu de la narration simpliste « plus de degré = plus de surapprentissage ». **Correction pédagogique réelle**, pas cosmétique.

**Math byte-grounded** : `ecart = train - test` cohérent (0,775-0,831=-0,056 ✓ etc.). `m=40` (train_size) corroboré par code cell 14 `train_test_split(...,train_size=40,random_state=42,stratify=...)`. `random_state=42` = déterministe (les valeurs sont reproductibles, pas aléatoires par run) → le fix est stable.

**Stop & Repair rule 6 honorée** : 1 seule cellule markdown (idx 15) changée, **0 cellule code touchée source/out/ec** (ec préservé base↔head sur les 25/25 cellules), cell-count 25=25, nbformat 4.5. Le fix corrige l'interp pour matcher un output **déjà committé et inchangé** (pas de re-exec nécessaire — l'output était correct, c'est la prose qui mentait).

**0 fake output** (0 output touché), **0 leak** (sk-/ghp_/gho_/AKIA/xox/`C:\Users`/`/home/`/`[A-Z]:\\` = 0 sur la cell changée, prose FR pure + valeurs numériques), **0 CJK** (HARD RULE Emerjesse respectée, node-regex `[一-鿿㐀-䶿]`).

△ **Nit non-bloquant** : la nouvelle prose cite « 2.5 recommandait la validation croisée » — cohérent (section 2.5 du cours), mais la dernière ligne (« une classe plus riche *finit par* généraliser moins bien ») reste légèrement en tension avec l'observation que deg 5 (plus riche que deg 1) généralise *mieux* (test 0,914 > 0,831). Le wording adouci « finit par » (vs l'ancien « généralise moins bien ») résout la tension, mais un lecteur pourrait la remarquer. Cosmétique, la pédagogie générale (dimension VC exige plus d'exemples pour les classes riches) reste correcte.

Audit-fix de qualité supérieure — corrige une interp demi-vérité en la rendant byte-exact aux sorties ET conceptuellement plus juste (surface l'inversion de signe de l'écart train-test = le vrai phénomène pédagogique ici), markdown-only rule 6, 0 leak 0 CJK. — **[NanoClaw]** (myia-ai-01)
